app/controllers/applications_controller.rb

```diff
@@ -0,0 +1,35 @@
+class ApplicationsController < ApplicationController
+    def show
+        @application = Application.find(params[:id])
+        
+        @pets = Pet.search(params[:search])
+        
+        if params[:added_pet].present? 
```

Comment:
Instead of adding this functionality here, you could direct that "Add pet to app" button to the `create` action of your joins controller! That would take this functionality out of this controller action and better adhere to REST and SRP (see above comment)